src/cli/self_update/windows.rs

```diff
@@ -285,6 +282,55 @@ pub fn do_remove_from_path() -> Result<()> {
     _apply_new_path(new_path)
 }
 
+const RUSTUP_UNINSTALL_ENTRY: &str = r"Software\Microsoft\Windows\CurrentVersion\Uninstall\Rustup";
+
+pub fn do_add_to_programs() -> Result<()> {
+    use std::path::PathBuf;
+
+    let key = RegKey::predef(HKEY_CURRENT_USER)
+        .create_subkey(RUSTUP_UNINSTALL_ENTRY)
+        .chain_err(|| ErrorKind::PermissionDenied)?
+        .0;
+
+    // Don't overwrite registry if Rustup is already installed
```

Comment:
I'm not certain this is right, but for now lets leave it. I think it might be more that we want to determine default RUSTUP_HOME being overridden etc, and otherwise we should be writing the current correct value into the registry each time.

Reply:
What would be the "current correct value"? As someone who overrides RUSTUP_HOME and use temporary installs of  rustup in parallel of the standard one, I wouldn't want those temporary installs to replace the standard one in the registry.

Reply:
A later install wont override a previous one, as long as the previous install is still installed.
However when uninstalling, the registry key will be unconditionally removed, and then recreated on the next install. See https://github.com/rust-lang/rustup/pull/2670#discussion_r592748920

By the way, any `rustup self uninstall` will unistall rustup in the current `RUSTUP_HOME`, regardless of where the executable you're running is.